sqlserver/tests/test_stored_procedures.py

```diff
@@ -0,0 +1,363 @@
+    return dbm_instance['tags']
+
+
+@pytest.fixture(autouse=True)
+def stop_orphaned_threads():
+    # make sure we shut down any orphaned threads and create a new Executor for each test
+    DBMAsyncJob.executor.shutdown(wait=True)
+    DBMAsyncJob.executor = ThreadPoolExecutor()
+
+
+@pytest.fixture
+def dbm_instance(instance_docker):
+    instance_docker['dbm'] = True
+    instance_docker['min_collection_interval'] = 1
+    instance_docker['query_metrics'] = {'enabled': False}
+    # set a very small collection interval so the tests go fast
+    instance_docker['procedure_metrics'] = {
+        'enabled': True,
+        'run_sync': True,
+        'collection_interval': 0.1,
+    }
+    return copy(instance_docker)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+@pytest.mark.parametrize(
+    "expected_columns,available_columns",
+    [
+        [
+            ["execution_count", "total_worker_time"],
+            ["execution_count", "total_worker_time"],
+        ],
+        [
+            ["execution_count", "total_worker_time", "some_missing_column"],
+            ["execution_count", "total_worker_time"],
+        ],
+    ],
+)
+def test_get_available_procedure_metrics_columns(dbm_instance, expected_columns, available_columns):
+    check = SQLServer(CHECK_NAME, {}, [dbm_instance])
+    check.initialize_connection()
+    _conn_key_prefix = "dbm-test-procedures"
+    with check.connection.open_managed_default_connection(key_prefix=_conn_key_prefix):
+        with check.connection.get_managed_cursor(key_prefix=_conn_key_prefix) as cursor:
+            result_available_columns = check.procedure_metrics._get_available_procedure_metrics_columns(
+                cursor, expected_columns
+            )
+            assert result_available_columns == available_columns
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_get_procedure_metrics_query_cached(aggregator, dbm_instance, caplog):
+    caplog.set_level(logging.DEBUG)
+    check = SQLServer(CHECK_NAME, {}, [dbm_instance])
+    check.initialize_connection()
+    _conn_key_prefix = "dbm-test-procedures"
+    with check.connection.open_managed_default_connection(key_prefix=_conn_key_prefix):
+        with check.connection.get_managed_cursor(key_prefix=_conn_key_prefix) as cursor:
+            for _ in range(3):
+                query = check.procedure_metrics._get_procedure_metrics_query_cached(cursor)
+                assert query, "query should be non-empty"
+    times_columns_loaded = 0
+    for r in caplog.records:
+        if r.message.startswith("found available sys.dm_exec_procedure_stats columns"):
+            times_columns_loaded += 1
+    assert times_columns_loaded == 1, "columns should have been loaded only once"
+
+
+test_procedure_metrics_parametrized = (
+    "database,query,param_groups,execution_count,expected_objects",
+    [
+        [
+            "master",  # database
+            "EXEC multiQueryProc",  # query
```

Comment:
did you mean to add these extra comments here?

Reply:
Yes